README.md

```diff
@@ -173,13 +187,28 @@ python train_reader.py \
 	--output_dir {path to output dir}
 ```
 
-Notes: 
+Notes:
 - if you use pytext_bert or fairseq_roberta, you need to download pre-trained weights and specify --pretrained_file parameter. Specify the dir location of the downloaded files for 'pretrained.fairseq.roberta-base' resource prefix for RoBERTa model or the file path for pytext BERT (resource name 'pretrained.pytext.bert-base.model').
-- Reader training pipeline does model validation every --eval_step batches 
+- Reader training pipeline does model validation every --eval_step batches
 - As the bi-encoder, it saves model checkpoints on every validation
 - Like the bi-encoder, there is no stop condition besides a specified amount of epochs to train.
 - Like the bi-encoder, there is no best checkpoint selection logic, so one needs to select that based on dev set validation performance which is logged in the train process output.
-- Our current code only calculates the Exact Match metric. 
+- Our current code only calculates the Exact Match metric.
+
+## Reader model inference
+
+In order to make an inference, run `train_reader.py` without specifying `train_file`. Make sure to specify `model_file` with the path to the checkpoint, `passages_per_question_predict` with number of passages per question (being used when saving the prediction file), and `eval_top_docs` with a list of numbers for number of passages per question which results to be printed. The example command line is as follows.
+
+```batch
+python train_reader.py \
+  --prediction_results_file {some dir}/results.json \
+  --eval_top_docs 10 20 40 50 80 100 \
+  --dev_file {path to data.retriever_results.nq.single.test file} \
```

Comment:
as of NQ, we also provide data.reader.nq.single.test resource which is supposed to be a preprocessed version of the data.retriever_results.nq.single.test but it somehow gives a bit better results.

Reply:
I don't think we should encourage people to use provided preprocessed data. Preprocessed reader data is technically the results from the dense retrieval. When they are re-training the model, they would want to reproduce the results without using preprocessed data. 

Also, it would be problematic if the code does not reproduce provided preprocessed data.